openassessment/xblock/submission_mixin.py

```diff
@@ -414,6 +414,20 @@ def upload_url(self, data, suffix=''):  # pylint: disable=unused-argument
         if 'contentType' not in data or 'filename' not in data:
             return {'success': False, 'msg': self._("There was an error uploading your file.")}
 
+        if not self.allow_multiple_files:
+
+            # Here we check if there are existing file uploads by checking for
+            # an existing download url for any of the upload slots.
+            # Note that we can't use self.saved_files_descriptions because that
+            # is populated before files are uploaded
+            for i in range(self.MAX_FILES_COUNT):
+                file_url = self._get_download_url(i)
+                if file_url:
+                    return {'success': False,
+                            'msg': self._(u"Only a single file upload is allowed for this assessment.")}
```

Comment:
nit directed at original author: we no longer need to use 'u' prefix.

Reply:
back when he wrote this, we did 😆